pkg/client/client.go

```diff
@@ -268,13 +270,30 @@ func (o *OptimizelyClient) decideForKeys(userContext OptimizelyUserContext, keys
 	if len(keys) == 0 {
 		return decisionMap
 	}
+	allOptions := o.getAllOptions(options)
 
-	enabledFlagsOnly := o.getAllOptions(options).EnabledFlagsOnly
-	for _, key := range keys {
-		optimizelyDecision := o.decide(userContext, key, options)
-		if !enabledFlagsOnly || optimizelyDecision.Enabled {
-			decisionMap[key] = optimizelyDecision
+	var userProfile *decision.UserProfile
+	ignoreUserProfileSvc := o.UserProfileService == nil || allOptions.IgnoreUserProfileService
+	if !ignoreUserProfileSvc {
+		up := o.UserProfileService.Lookup(userContext.GetUserID())
+		if up.ID == "" {
+			up = decision.UserProfile{
+				ID:                  userContext.GetUserID(),
+				ExperimentBucketMap: map[decision.UserDecisionKey]string{},
+			}
 		}
+		userProfile = &up
+		userContext.userProfile = userProfile
+	}
+
+	for _, key := range keys {
+		optimizelyDecision := o.decide(&userContext, key, options)
```

Comment:
Would it be good to add `optimizelyDecision, err` handling here?

Reply:
Actually `decide` doesn't return error